sys/slot_util/slot_util.c

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright (C) 2017 Kaspar Schleiser <kaspar@schleiser.de>
+ *                    Inria
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     sys_slot_util
+ * @{
+ *
+ * @file
+ * @brief       Slot management functions
+ *
+ * @author      Kaspar Schleiser <kaspar@schleiser.de>
+ * @author      Francisco Acosta <francisco.acosta@inria.fr>
+ *
+ * @}
+ */
+#include <string.h>
+
+#include "cpu.h"
+#include "slot_util.h"
+#include "riot_hdr.h"
+
+#ifdef RIOT_VERSION
+#ifndef BOARD_NATIVE
+/*
+ * Store the start addresses of each slot.
+ * Take into account that CPU_FLASH_BASE represents the starting
+ * address of the bootloader, thus we skip it.
+ */
+const uint32_t slot_util_slots[] = {
```

Comment:
Same here as the documentation, this should not be `uint32_t` but a pointer type.

Reply:
It could even simply be defined as `riot_hdr_t *` as it is what it is supposed to be.